frontend/src/Frontend/Page/Transaction.hs

```diff
@@ -54,77 +57,104 @@ transactionPage
   => NetId
   -> ChainId
   -> BlockPayloadWithOutputs
+  -> BlockHeader
   -> m ()
-transactionPage netId cid bp = do
+transactionsPage netId cid bp bh = do
     let txs = _blockPayloadWithOutputs_transactionsWithOutputs bp
+        bhash = _blockHeader_hash bh
     el "h2" $ text $ (tshow $ length txs) <> " Transactions"
-    divClass "ui accordion" $ do
-      forM_ txs $ \(t, tout) -> mdo
-        open <- toggle False $ domEvent Click e
+    elClass "table" "ui celled table" $ do
+      el "thead" $ do
+        el "tr" $ do
+          el "th" $ text "Transaction"
+      el "tbody" $ do
+        forM_ txs $ \(t, _) -> do
+          let route = addNetRoute netId (unChainId cid) $ Chain_BlockHash :/ (hashB64U bhash, Block_Transaction :/ (T.pack $ show $ _transaction_hash t))
+          el "tr" $ el "td" $ case _pactCommand_payload $ _transaction_cmd t of
+            ExecPayload exec ->  routeLink route $ text $ _exec_code exec
+            ContPayload cont ->  routeLink route $ text $ _cont_pactId cont
+
+transactionPage
+  :: ( MonadApp r t m
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , Prerender js t m
+     , Routed t Text m
+     )
+  => NetId
+  -> ChainId
+  -> BlockPayloadWithOutputs
+  -> m ()
+transactionPage netId cid bp = do
+  dynTxHash :: Dynamic t Text <- askRoute
+  let txs = M.fromList $ map (\(t, tout) -> ((T.pack $ show $ _transaction_hash t), (t, tout))) $  _blockPayloadWithOutputs_transactionsWithOutputs bp
```

Comment:
This use of `show` is resulting in `Hash {unHash = ...` showing up in the URL.  Let's use the base64 encoding of the `Hash` data structure instead.